### ISSUE TITLE
Pre-training Observation: Model Reproducing Reference Images Despite Low Loss

### ISSUE THREAD

**Issue by wyhlovecpp** [stepfun-ai/Step1X-Edit#116]
Thanks for your great work. I've been reviewing past issues, and I saw your comments regarding your pre-training process, specifically the multi-stage training approach and the total training budget of 1 million steps. I also noted the fixed learning rate of 1e-5 and the mention of initial issues like ghosting artifacts.

I'm currently trying to replicate your pre-training setup. I'm training a model at a resolution of 1024x1024 with a batch size of 32, using your suggested learning rate of 1e-5. However, I've encountered an issue: after approximately 6000 steps, my loss has dropped significantly, yet the generated images still strongly reproduce the reference images, and are quite blurry.

I'd like to ask:

Did you experience similar behavior during your initial training stages, where the model reproduced the reference image despite low loss?

Roughly how long (in steps or iterations) did it take for your model to move past this initial "reproduce the reference" stage and start generating novel outputs, particularly in the first stage of training?

Any insights you could provide would be greatly appreciated.

<img width="471" height="251" alt="Image" src="https://github.com/user-attachments/assets/2976246d-6b09-4aef-9018-07f6ea9aa276" />

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/03cc7554-ba9d-4e26-8f35-92f9c2b0d287" />

Thanks!

**Comment by wyhlovecpp** [comment]
> I am also having such problem, the model simply copies the source image and ignore the text embedding. I also tried the reweighting strategy, but still did not work.
@veroveroxie  Have you considered whether this could be related to the positional encoding scheme? If the model uses absolute or poorly disentangled positional encodings, it might encourage overfitting to spatial layout and copying behavior instead of learning true. I think in the beginning, the reference image and target noisy image must use the same PE and don't use the text embedding. After a lot of training steps, the model can generate the image with the reference. And then let the text embedding to train together.

**Comment by veroveroxie** [comment]
I agree with you that it could be the positional encoding issue. Kontext use similar position encoding although they set first axis value to 1. The problem is that the model can still easily ignore the injected text embedding in your proposed second stage.  I tried many solutions but none works. Did you get a workaround for that?
